py/server/deephaven/table_factory.py

```diff
@@ -301,18 +268,38 @@ def delete(self, table: Table) -> None:
             DHError
         """
         try:
-            if not self.key_columns:
-                raise PermissionError("deletion on an append-only input table is not allowed.")
             self.j_input_table.delete(table.j_table)
         except Exception as e:
             raise DHError(e, "delete data in the InputTable failed.") from e
 
+    def get_key_names(self) -> List[str]:
+        """Gets the names of the key columns.
+
+        Returns:
+            a list with the names of the key columns of this input table
```

Comment:
```suggestion
    @property
    def get_key_names(self) -> List[str]:
        """The names of the key columns of the InputTable.
```

Reply:
I'm not very familiar w/ Python `@property`, but it looks like this method should now be called `key_names`, not `get_key_names`. Changing.